src/diff/props.js

```diff
@@ -39,39 +38,28 @@ const XLINK_NS = 'http://www.w3.org/1999/xlink';
  * @param {boolean} isSvg Whether or not this DOM node is an SVG node or not
  */
 function setProperty(dom, name, value, oldValue, isSvg) {
+	let v;
+
 	name = isSvg ? (name==='className' ? 'class' : name) : (name==='class' ? 'className' : name);
 
 	if (name==='style') {
+		// Always clear the previous styles
+		dom.style.cssText = EMPTY_OBJ;
 
-		/* Possible golfing activities for setting styles:
-		 *   - we could just drop String style values. They're not supported in other VDOM libs.
-		 *   - assigning to .style sets .style.cssText - TODO: benchmark this, might not be worth the bytes.
-		 *   - assigning also casts to String, and ignores invalid values. This means assigning an Object clears all styles.
-		 */
-		let s = dom.style;
+		for (let i in value) {
+			v = value[i];
+			v = typeof v==='number' && IS_NON_DIMENSIONAL.test(i)===false ? (v + 'px') : v;
 
-		if (typeof value==='string') {
-			s.cssText = value;
-		}
-		else {
-			if (typeof oldValue==='string') s.cssText = '';
-			else {
-				// remove values not in the new list
-				for (let i in oldValue) {
```

Comment:
Since the styles are cleared, like it was before, but now the _entire_ `cssText` gets removed, we don't need to go over the _old_ styles.

Reply:
this is really expensive though. it means changing one style property changes all style properties.

Reply:
Yeah. That's a great point. I'll redo this PR.